expr/functions/integralByInterval/function.go

```diff
@@ -55,7 +54,7 @@ func (f *integralByInterval) Do(ctx context.Context, e parser.Expr, from, until
 		current := 0.0
 		currentTime := arg.StartTime
 
-		name := fmt.Sprintf("integralByInterval(%s,'%s')", arg.Name, e.Args()[1].StringValue())
+		name := "integralByInterval(" + arg.Name + ",'" + e.Args()[1].StringValue() + "')"
```

Comment:
Replace `e.Args()[1]` with Arg(1) and move outside a cycle. It's old artifact (not cleaned). See
https://github.com/grafana/carbonapi/blob/9eb0a9e50888933515bb9756dbdd8954db423cf1/pkg/parser/parser.go#L111
Like

argStr := e.Arg(1).StringValue()
for j, arg := range args {
		name := "integralByInterval(" + arg.Name + ",'" + argStr + "')"

It's save some allocations during interface unboxing.

Reply:
Interesting. Thank you for the insight. I just noticed `intervalString` is defined in line 46, so I'll use that for this case.